### PR TITLE
[IMPROVEMENT] Add image loader

### LINKED ISSUE #794: Show loading indicator on images

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
@RocketChat/ReactNative

Closes #794

### CODE REVIEW

File: app/containers/message/Image.js [hunk 1]
```diff
@@ -55,6 +59,21 @@ export default class extends Component {
...
 
+	setLoading = (loading) => {
+		this.setState({ loading });
+	}
+
+	renderLoading() {
+		const { loading } = this.state;
+		if (loading) {
+			return (
+				<View style={styles.loading}>
```
Comment (IlarionHalushka) [inline app/containers/message/Image.js]: I suppose wrapping ActivityIndicator in View is not necessary here

Comment (pranavpandey1998official) [reply app/containers/message/Image.js]: It is to centre it.

Comment (IlarionHalushka) [reply app/containers/message/Image.js]: yeah, but you can remove View and add styles.loading to ActivityIndicator directly and see that nothing has changed :)

**Comment by pranavpandey1998official** [comment]
@diegolmello can you explain more on how it is not working properly on photoModal

**Comment by diegolmello** [comment]
@pranavpandey1998official it seems PhotoModal isn't loading images from cache.

**Comment by pranavpandey1998official** [comment]
@diegolmello when i tested it is loading from cache .. and i don't think its a good idea to enable loading on avatars as , even when fastImage is loading from cache it is showing that loading for split second and it doesn't look good to see whole bunch of loading on each and every avatar

**Comment by diegolmello** [comment]
@pranavpandey1998official Can you make a gif with that avatar loading?
I was talking about PhotoModal cache. If you tap in the image, will it load from cache?

**Comment by pranavpandey1998official** [comment]
@diegolmello it is cahing in photomodel too https://drive.google.com/file/d/1BEPY2hNAnBBQfPsr_2jlGI8Dgul1gasK/view?usp=sharing

**Comment by pranavpandey1998official** [comment]
this is the avtaar behaviour https://drive.google.com/file/d/1XHqSA0snQ-Zr-1XOOufpbihQBV10GTbc/view?usp=sharing

**Comment by diegolmello** [comment]
@pranavpandey1998official Ok then. Remove it from avatar and leave PhotoModal as it is.

**Comment by pranavpandey1998official** [comment]
@diegolmello done

**Pull request closed without merging**